### PR TITLE
Fix LibP2P-Daemon installation in setup.py

### PR DESCRIPTION
**Pull request by deniskamazur**
* Rename `ProtoCompileInstall` and `ProtoCompileDevelop` to `Install` and `Develop`
* Install LibP2P-Daemon on setup install and setup develop
* Install Golang in Circle CI builds
* Add p2pd binary to gitignore

### CODE REVIEW

**Review by justheuristic — APPROVED** [review]
Good point, let's use binaries in install mode, we just need to figure out how best to store them

**Comment by justheuristic** [comment]
:note from @mryab :
- check version when downloading (i.e. not just latest)
- do not re-download p2pd if control sum matches the reference

File: setup.py [hunk 1]
```diff
@@ -49,46 +50,54 @@ def proto_compile(output_path):
...
-            status = os.system(f'go build -o {os.path.join(here, "hivemind/hivemind_cli", "p2pd")}')
-            if status:
-                raise RuntimeError('Failed to build or install libp2p-daemon:'\
-                                   f' exited with status code :{status}')
-
-
-class ProtoCompileInstall(install):
+def install_libp2p_daemon(build=False):
+    # verify golang installation if build flag is specified
+    if build:
```
Comment (mryab) [inline setup.py:55]: In essence, you have two functions with entirely different code. It's best to separate them into two functions with different names

**Review by mryab — CHANGES_REQUESTED** [review]

File: setup.py [hunk 2]
```diff
@@ -30,6 +35,14 @@ def __exit__(self, etype, value, traceback):
         os.chdir(self.savedPath)
 
 
+def md5(fname):
+    hash_md5 = hashlib.md5()
+    with open(fname, "rb") as f:
+        for chunk in iter(lambda: f.read(4096), b""):
```
Comment (mryab) [inline setup.py:41]: Move 4096 to a constant or at least an argument with default value

**Review by mryab — CHANGES_REQUESTED** [review]

**Review by mryab — APPROVED** [review]

File: setup.py [hunk 3]
```diff
@@ -67,38 +79,44 @@ def install_libp2p_daemon():
...
+                raise RuntimeError('Failed to build or install libp2p-daemon:'
                                    f' exited with status code :{status}')
 
 
-class ProtoCompileInstall(install):
+def libp2p_download_install():
+    install_path = os.path.join(here, 'hivemind/hivemind_cli/')
+    binary_path = os.path.join(install_path, 'p2pd')
+    if 'p2pd' not in os.listdir(install_path) or md5(binary_path) != P2PD_CHECKSUM:
+        print('Downloading Peer to Peer Daemon')
```
[the remark this answers is not in the document]
Comment (mryab) [reply setup.py:85]: Will fix that when merging to master

**Pull request merged**